spec/importer_spec.rb

```diff
@@ -0,0 +1,191 @@
+RSpec.describe Ronin::Masscan::Importer do
+  let(:masscan_path) { File.expand_path(File.join(__dir__, '..', 'spec', 'fixtures', 'masscan.json')) }
+  let(:output_file)  { Masscan::OutputFile.new(masscan_path) }
+
+  before(:all) do
+    Ronin::DB.connect('sqlite3::memory:')
+  end
+
+  after(:all) do
+    Ronin::DB::IPAddress.destroy_all
+    # Ronin::DB::Port.destroy_all
+    Ronin::DB::OpenPort.destroy_all
+  end
+
+  describe '.import_file' do
+    let(:expected_values) do
+      [
+        Ronin::DB::IPAddress,
+        Ronin::DB::Port,
+        Ronin::DB::OpenPort,
+        Ronin::DB::IPAddress,
+        Ronin::DB::Port,
+        Ronin::DB::OpenPort,
+        Ronin::DB::IPAddress
+      ]
+    end
+
+    it 'must import records from file' do
+      yielded_values = []
+
+      subject.import_file(masscan_path) do |value|
+        yielded_values << value
+      end
+
+      expect(yielded_values.size).to eq(7)
+      expect(yielded_values.map(&:class)).to match(expected_values)
+    end
+  end
+
+  describe '.import' do
+    let(:expected_values) do
+      [
+        Ronin::DB::IPAddress,
+        Ronin::DB::Port,
+        Ronin::DB::OpenPort,
+        Ronin::DB::IPAddress,
+        Ronin::DB::Port,
+        Ronin::DB::OpenPort,
+        Ronin::DB::IPAddress
+      ]
+    end
+
+    context 'when an unknown record type is encountered' do
+      before do
+        allow(output_file).to receive(:each).and_yield("unknown")
+      end
+
+      it 'must raise NotImplementedError' do
+        expect { |b|
+          subject.import(output_file, &b)
+        }.to raise_error(NotImplementedError)
+      end
+    end
+
+    context 'when a block is given' do
+      it 'must yield imported records' do
+        yielded_values = []
+
+        subject.import_file(masscan_path) do |value|
+          yielded_values << value
+        end
+
+        expect(yielded_values.size).to eq(7)
+        expect(yielded_values.map(&:class)).to match(expected_values)
+      end
+    end
+
+    context 'when no block is given' do
+      it 'must return imported records' do
+        expect(subject.import(output_file).size).to eq(7)
+        expect(subject.import(output_file).map(&:class)).to match(expected_values)
+      end
+    end
+  end
+
+  describe '.import_status' do
+    let(:status1) { instance_double('Masscan::Status', status: :open, protocol: :icmp, ip: ip_addr) }
+    let(:status2) { instance_double('Masscan::Status', status: :open, protocol: :tcp) }
+    let(:ip_addr) { IPAddr.new('1.1.1.1') }
+
+    context 'when status is :open' do
+      context 'when protocol is :icmp' do
+        it 'must import ip address' do
+          expect(subject).to receive(:import_ip_address).with(ip_addr)
+
+          subject.import_status(status1)
+        end
+      end
+
+      context 'when protocol is not :icmp' do
+        it 'must import open port' do
+          expect(subject).to receive(:import_open_port_status).with(status2)
+
+          subject.import_status(status2)
```

Comment:
`import_status` also needs to be fully tested to verify it yields/returns the appropriate DB records. This will prevent the specs from becoming brittle and break if we every change the internal logic of `import_status`.